modules/k8s/kubectl.go

```diff
@@ -0,0 +1,101 @@
+func RunKubectl(t *testing.T, options *KubectlOptions, args ...string) {
+	err := RunKubectlE(t, options, args...)
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+// RunKubectlE will call kubectl using the provided options and args.
+func RunKubectlE(t *testing.T, options *KubectlOptions, args ...string) error {
+	_, err := RunKubectlAndGetOutputE(t, options, args...)
+	return err
+}
+
+// RunKubectlAndGetOutputE will call kubectl using the provided options and args, returning the output of stdout and
+// stderr.
+func RunKubectlAndGetOutputE(t *testing.T, options *KubectlOptions, args ...string) (string, error) {
+	cmdArgs := []string{}
+	if options.ContextName != "" {
+		cmdArgs = append(cmdArgs, "--context", options.ContextName)
+	}
+	if options.ConfigPath != "" {
+		cmdArgs = append(cmdArgs, "--kubeconfig", options.ConfigPath)
+	}
+	cmdArgs = append(cmdArgs, args...)
+	command := shell.Command{
+		Command: "kubectl",
+		Args:    cmdArgs,
+		Env:     options.Env,
+	}
+	return shell.RunCommandAndGetOutputE(t, command)
+}
+
+// KubectlDeleteFromString will take in a kubernetes resource config as a string and delete it on the cluster specified
+// by the provided kubectl options.
+func KubectlDeleteFromString(t *testing.T, options *KubectlOptions, configData string) {
+	require.NoError(t, KubectlDeleteFromStringE(t, options, configData))
+}
+
+// KubectlDeleteFromStringE will take in a kubernetes resource config as a string and delete it on the cluster specified
+// by the provided kubectl options. If it fails, this will return the error.
+func KubectlDeleteFromStringE(t *testing.T, options *KubectlOptions, configData string) error {
+	tmpfile, err := StoreConfigToTempFileE(t, configData)
+	if err != nil {
+		return err
+	}
+	defer os.Remove(tmpfile)
+	return RunKubectlE(t, options, "delete", "-f", tmpfile)
+}
+
+// KubectlApplyFromString will take in a kubernetes resource config as a string and apply it on the cluster specified
+// by the provided kubectl options.
+func KubectlApplyFromString(t *testing.T, options *KubectlOptions, configData string) {
+	require.NoError(t, KubectlApplyFromStringE(t, options, configData))
+}
+
+// KubectlApplyFromStringE will take in a kubernetes resource config as a string and apply it on the cluster specified
+// by the provided kubectl options. If it fails, this will return the error.
+func KubectlApplyFromStringE(t *testing.T, options *KubectlOptions, configData string) error {
+	tmpfile, err := StoreConfigToTempFileE(t, configData)
+	if err != nil {
+		return err
+	}
+	defer os.Remove(tmpfile)
+	return RunKubectlE(t, options, "apply", "-f", tmpfile)
+}
+
+// StoreConfigToTempFile will store the provided config data to a temporary file created on the os and return the
+// filename.
+func StoreConfigToTempFile(t *testing.T, configData string) string {
+	out, err := StoreConfigToTempFileE(t, configData)
+	require.NoError(t, err)
```

Comment:
That is cleaner. Will make mental note to use this elsewhere in Terratest in the future.